generate/sre-authorization/osd-sre-admin.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+
+BLACKLIST=`cat blacklist.json`
+
+echo "apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: osd-sre-admin
+rules:"
+
+# /api
+GROUP=""
+for VERSION in `oc get --raw /api | jq -r .versions[] | sort`;
+do
+    # if the group isn't blacklisted just allow everything
+    COUNT=`echo $BLACKLIST | jq ".[] | select(.apiGroup == \"$GROUP\")" | wc -l`
+
+    if [ "$COUNT" == "0" ];
+    then
+        # not blacklisted, continue
+        echo "- apiGroups:"
+        echo "  - \"\""
+        echo "  resources:"
+        echo "  - '*'"
+        echo "  verbs:"
+        echo "  - '*'"
+        continue
+    fi
+
+    for RESOURCE in `oc get --raw /api/${VERSION} | jq -r .resources[].name | sort`;
+    do
+        # something was blacklisted, see if it was the resource
+        COUNT=`echo $BLACKLIST | jq ".[] | select(.apiGroup == \"$GROUP\") | select(.resources[] == \"$RESOURCE\")" | wc -l`
+
+        if [ "$COUNT" == "0" ];
+        then
+            # not blacklisted, continue
+            echo "- apiGroups:"
+            echo "  - \"\""
+            echo "  resources:"
+            echo "  - $RESOURCE"
+            echo "  verbs:"
+            echo "  - '*'"
+            continue
+        fi
+
+        VERBS=""
+        VERB_COUNT=0
+
+        for VERB in `oc get --raw /api/${VERSION} | jq -r ".resources[] | select(.name == \"$RESOURCE\") | .verbs[]" | sort`;
```

Comment:
if I do a `oc get --raw /api/v1` or `oc get --raw /apis`, custom objects like prometheusrules never appear.

Should we consider another way to tackle those objects? Like `oc get crds`

Reply:
nevermind, I just saw the step below